src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -125,10 +133,10 @@ def ruby_response(response)
   end
 
   def create_client_if_nil
-    if @client.nil?
+    if @@client.nil?
```

Comment:
If you use the `||=` in the `self.client` method above, then this nil check can go away.  The `||=` pattern is much more idiomatic in Rubyland.